gobblin-data-management/src/main/java/org/apache/gobblin/data/management/source/LoopingDatasetFinderSource.java

```diff
@@ -85,31 +86,52 @@ public WorkUnitStream getWorkunitStream(SourceState state) {
       int maxWorkUnits = state.getPropAsInt(MAX_WORK_UNITS_PER_RUN_KEY, MAX_WORK_UNITS_PER_RUN);
 
       List<WorkUnitState> previousWorkUnitStates = state.getPreviousWorkUnitStates();
-      Optional<WorkUnitState> maxWorkUnit;
-      try {
-        maxWorkUnit = previousWorkUnitStates.stream().reduce((wu1, wu2) -> {
-          int wu1Ordinal = wu1.getPropAsInt(WORK_UNIT_ORDINAL);
-          int wu2Ordinal = wu2.getPropAsInt(WORK_UNIT_ORDINAL);
-          return wu1Ordinal > wu2Ordinal ? wu1 : wu2;
-        });
-      } catch (NumberFormatException nfe) {
-        throw new RuntimeException("Work units in state store are corrupted! Missing or malformed " + WORK_UNIT_ORDINAL);
+      int tempWorkUnitOrdinal = 0;
+      Optional<WorkUnitState> optionalMaxWorkUnit = Optional.empty();
+      List<WorkUnit> failedPreviousWorkUnits = new ArrayList<>();
+      int maxRetryTimes = state.contains(ConfigurationKeys.MAX_TASK_RETRIES_KEY) ? state
```

Comment:
I don't think you should use task_max_retries, as that configuration is intended for retries within a job. Create a new one instead.